addons/hr_holidays/models/hr_leave.py

```diff
@@ -1591,13 +1591,14 @@ def _get_responsible_for_approval(self):
         elif self.validation_type == 'hr' or (self.validation_type == 'both' and self.state == 'validate1'):
             if self.holiday_status_id.responsible_ids:
                 responsible = self.holiday_status_id.responsible_ids
-            else:
-                responsible = self.env.ref('hr_holidays.group_hr_holidays_user').users.filtered(lambda u: self.holiday_status_id.company_id in u.company_ids)
+
         return responsible
 
     def activity_update(self):
         to_clean, to_do = self.env['hr.leave'], self.env['hr.leave']
         activity_vals = []
+        mail_act_leave_approval_xml_id = self.env.ref('hr_holidays.mail_act_leave_approval').id
+        model_hr_leave_id = self.env.ref('hr_holidays.model_hr_leave').id
```

Comment:
Should be moved before the `user_ids = holiday.sudo()._get_responsible_for_approval().ids or self.env.user.ids` to not fetch the records even they are not needed (that's why the staging failed I think) - same thing below